csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -938,26 +939,12 @@ internal bool isLocal(Reference r)
             {
                 checkForDeactivation();
 
-                //
-                // Proxies which have at least one endpoint in common with the
-                // endpoints used by this object adapter's incoming connection
-                // factories are considered local.
-                //
-                for (int i = 0; i < endpoints.Length; ++i)
+                // Proxies which have at least one endpoint in common with the published endpoints are considered local.
+                foreach (EndpointI endpoint in endpoints)
                 {
-                    foreach (EndpointI endpoint in _publishedEndpoints)
-                    {
-                        if (endpoints[i].equivalent(endpoint))
-                        {
-                            return true;
-                        }
-                    }
-                    foreach (IncomingConnectionFactory factory in _incomingConnectionFactories)
```

Comment:
I simplified the "isLocal" check (used for collocation optimization) - with this update, we check only the published endpoints.

Previously, we checked the published endpoints and also the "published endpoint" of each incoming connection factory, which I find very much over the top.